### PR TITLE
feature: LVGL display flush offload

### LINKED ISSUE #57705: LVGL port should not block rendering thread in display flush callback
**Is your enhancement proposal related to a problem? Please describe.**
Zephyr's LVGL port currently calls `display_write` from the display flush callback. This API blocks the calling thread until the display framebuffer is written. Since LVGL calls the flush callback from the rendering thread, this results in the rendering itself being blocked while the framebuffer is written to the display.

**Describe the solution you'd like**
Our LVGL port should ideally do the following:
- spawn a background thread or workqueue at init
- in the display flush callback, submit the provided LVGL buffer to the workqueue
- the workqueue should call `display_write`, then call `lv_disp_flush_ready` when the write is completed.
- we should additionally provide a `wait_cb` to LVGL, that waits for the workqueue to call `lv_disp_flush_ready` (this can be achieved with a semaphore)

**Describe alternatives you've considered**
We could modify the display API to add some form of async API, which takes a callback to notify of render completion. This method would potentially complicate the display API, and LVGL would need a way to fallback to the traditional `display_write` API if there was no ASYNC API available.

### PR DESCRIPTION
**Pull request by danieldegrasse**
Enable offloading of display_write call to background thread for color displays. This feature is opt-in, as it may offer significant performance gains for every display pipeline.

When enabled display_write and lv_disp_flush_ready will be called from a background thread. This means that while the display driver waits on the hardware to render the framebuffer, the LVGL rendering thread will not be blocked.

`CONFIG_LV_Z_FLUSH_THREAD=n`: 72 FPS
`CONFIG_LV_Z_FLUSH_THREAD=y`: 154 FPS

Note: this PR is an implementation of https://github.com/zephyrproject-rtos/lvgl/pull/40 that has been updated now that LVGL gluecode is within Zephyr proper

### CODE REVIEW

**Review by pdgendt — DISMISSED** [review]
I would argue to enable it by default, but it's fine either way.
Thanks!

**Comment by danieldegrasse** [comment]
> I would argue to enable it by default, but it's fine either way.
Thanks!

I want to keep this feature as opt in primarily because it won't work with the monochrome display port, where `display_write` can be called twice. For color displays, especially those using LCD controllers, this feature makes sense in most cases.

**Comment by danieldegrasse** [comment]
@jakkra FYI

**Comment by jakkra** [comment]
> @jakkra FYI- this PR has been updated to apply the changes to the LVGL gluecode that is now within the Zephyr tree, and no longer updates the west manifest

Great! One question, where did this go https://github.com/zephyrproject-rtos/lvgl/blob/7b6acfdb03019154500665ce8c919f4d4f88f8e4/zephyr/lvgl_display.c#L110?

**Comment by danieldegrasse** [comment]
> Great! One question, where did this go [zephyrproject-rtos/lvgl@7b6acfd/zephyr/lvgl_display.c#L110](https://github.com/zephyrproject-rtos/lvgl/blob/7b6acfdb03019154500665ce8c919f4d4f88f8e4/zephyr/lvgl_display.c#L110)?

Evidently got lost while moving the changes over- glad I pinged you! Should be there now

File: modules/lvgl/Kconfig [hunk 1]
```diff
@@ -53,6 +54,29 @@ endchoice
 config LV_COLOR_16_SWAP
 	bool
 
+config LV_Z_FLUSH_THREAD
+	bool "Flush LVGL frames in a separate thread"
+	help
```
Comment (dleach02) [inline modules/lvgl/Kconfig:59]: Why wouldn't we want to set a default true here for this option?

Comment (jakkra) [reply modules/lvgl/Kconfig:59]: This was asked in the original PR here https://github.com/zephyrproject-rtos/zephyr/pull/59166#issuecomment-1590193173

**Review by pdgendt — APPROVED** [review]

**Review by faxe1008 — APPROVED** [review]
Thanks for resubmitting against the main repo. Tested this on a mimxrt1060, no issues found.

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**